debug/README.md

```diff
@@ -0,0 +1,40 @@
+# gNOI `Debug` Streaming RPC Design
+
+**Contributors**: hines@google.com, robjs@google.com
+**Last Updated**: 2023-11-04
+
+## Background
+
+* [gNOI Repository](https://github.com/openconfig/gnoi)
+* [gNOI `Debug` service](https://github.com/openconfig/gnoi/tree/master/debug)
+
+For all legacy devices that provided a CLI on box, providers  have leveraged this CLI through services to provide users with the ability to interact with device CLI's to both get information as well as set operational state on devices.  This interaction is very vendor specific and requires significant overhead to maintain the vendor specific bindings throughout the operational life cycle of a device.
+
+With the introduction of g* services, the goal has been to remove vendor specific data formats from the view operators.  This lets operators have standard models for interacting with any number of vendor devices consistently. There however are gaps between those endpoints versioning and the ability to troubleshoot specific data on a device before API's can be updated.  This proposal is to enable a lightweight interface via gRPC to still access shell level interactions on the device in a secure, maintainable way.
+
```

Comment:
Suggest to add here:

"It is expected that where structured data is identified to be useful for an operator the suite of g* services, and/or underlying data models (e.g., OpenConfig) are extended to cover these use cases. However, it is critical to recognise that these APIs are not the "native" device APIs, and hence upgrading them requires some new software to be introduced."